source/kernel/src/trace.rs

```diff
@@ -82,29 +99,116 @@ impl SerialCollector {
             }
         };
         wgr.commit(len);
+        self.in_send.store(false, Ordering::Release);
 
         // return true if we committed a non-zero number of bytes.
         len > 0
     }
 
-    async fn worker(rx: bbq::Consumer, port: serial_mux::PortHandle) {
-        loop {
-            let rgr = rx.read_grant().await;
+    async fn worker(
+        &'static self,
+        rx: bbq::Consumer,
+        port: serial_mux::PortHandle,
+        k: &'static crate::Kernel,
+    ) {
+        use futures::FutureExt;
+        use maitake::time;
+        use postcard::accumulator::{CobsAccumulator, FeedResult};
+
+        // we probably won't use 256 whole bytes of cobs yet since all the host
+        // -> target messages are quite small
+        let mut cobs_buf: CobsAccumulator<16> = CobsAccumulator::new();
```

Comment:
We're gunna want some kind of config soon.

Reply:
definitely, yeah